files_opds/ajax/personal.php

```diff
@@ -12,39 +12,40 @@
 
 namespace OCA\Files_Opds;
 
-\OCP\JSON::callCheck();
-\OCP\JSON::checkLoggedIn();
+\OC_JSON::callCheck();
+\OC_JSON::checkLoggedIn();
 
 $l = \OC::$server->getL10N('files_opds');
 
 $opdsEnable = isset($_POST['opdsEnable']) ? $_POST['opdsEnable'] : 'false';
 $rootPath = isset($_POST['rootPath']) ? $_POST['rootPath'] : '/Library';
 $fileTypes = isset($_POST['fileTypes']) ? $_POST['fileTypes'] : '';
 $skipList = isset($_POST['skipList']) ? $_POST['skipList'] : 'metadata.opf,cover.jpg';
-$feedTitle = isset($_POST['feedTitle']) ? $_POST['feedTitle'] : $l->t("%s's Library", \OCP\User::getDisplayName());
+$feedTitle = isset($_POST['feedTitle']) ? $_POST['feedTitle'] : $l->t("%s's Library", \OC_User::getDisplayName());
+$opdsoldMime = isset($_POST['opdsoldMime']) ? $_POST['opdsoldMime'] : '';
 
 if (!strlen($rootPath) ||
-    \OC\Files\Filesystem::isValidPath($rootPath) === false || 
+    \OC\Files\Filesystem::isValidPath($rootPath) === false ||
     \OC\Files\Filesystem::file_exists($rootPath) === false ) {
-	\OCP\JSON::error(
+	\OC_JSON::error(
 		array(
 			'data' => array('message'=> $l->t('Directory does not exist!'))
 		)
 	);
 } else {
 	Config::set('root_path', $rootPath);
-        Config::set('enable', $opdsEnable);
-        Config::set('file_types', $fileTypes);
-        Config::set('skip_list', $skipList);
-        Config::set('feed_title', $feedTitle);
+	Config::set('enable', $opdsEnable);
+	Config::set('file_types', $fileTypes);
+	Config::set('skip_list', $skipList);
+	Config::set('feed_title', $feedTitle);
+	Config::set('old_mime', $opdsoldMime);
 	Config::set('id', Util::genUuid());
 
-       	\OCP\JSON::success(
+       	\OC_JSON::success(
```

Comment:
This works for now, but it is internal API, which might break at any point in time. Best is to move away from this style over to a app framework based controller, that can accomplish the same: https://docs.nextcloud.com/server/stable/developer_manual/app/requests/controllers.html